pywinauto/base_application.py

```diff
@@ -576,64 +577,117 @@ def _ctrl_identifiers(self):
+                if current_elem is None:
+                    elem_node = ElementTreeNode(None, current_id, [])
+                    current_elem_parent_children.append(elem_node)
+                else:
+                    if current_node_depth <= depth:
+                        if current_elem_parent_children is not None:
+                            current_id += 1
+                            elem_node = ElementTreeNode(current_elem, current_id, [])
+                            current_elem_parent_children.append(elem_node)
+                            element_list.append(current_elem)
+                        else:
+                            elem_node = root_node
+                        child_elements = current_elem.children()
+                        if len(child_elements) > max_width and current_node_depth < depth:
+                            elem_stack.append((None, elem_node.children, current_node_depth + 1))
+                            width_limit_reached = True
+                        for i in range(min(len(child_elements) - 1, max_width - 1), -1, -1):
+                            elem_stack.append((child_elements[i], elem_node.children, current_node_depth + 1))
+                    else:
+                        depth_limit_reached = True
+            return root_node, depth_limit_reached, width_limit_reached
+
+        # Create a list of this control, all its descendants
+        all_ctrls = [this_ctrl]
+
+        # Build element tree
+        elements_tree, depth_limit_reached, width_limit_reached = create_element_tree(all_ctrls)
+
+        show_best_match_names = self.allow_magic_lookup and not (depth_limit_reached or width_limit_reached)
+        if show_best_match_names:
+            # Create a list of all visible text controls
+            txt_ctrls = [ctrl for ctrl in all_ctrls if ctrl.can_be_label and ctrl.is_visible() and ctrl.window_text()]
+
+            # Build a dictionary of disambiguated list of control names
+            name_ctrl_id_map = findbestmatch.UniqueDict()
+            for index, ctrl in enumerate(all_ctrls):
+                ctrl_names = findbestmatch.get_control_names(ctrl, all_ctrls, txt_ctrls)
+                for name in ctrl_names:
+                    name_ctrl_id_map[name] = index
+
+            # Swap it around so that we are mapped off the control indices
+            ctrl_id_name_map = {}
+            for name, index in name_ctrl_id_map.items():
+                ctrl_id_name_map.setdefault(index, []).append(name)
+
+        def print_identifiers(element_node, current_depth=0, log_func=print):
             """Recursively print ids for ctrls and their descendants in a tree-like format"""
-            if len(ctrls) == 0 or current_depth > depth:
-                return
+            if current_depth == 0:
+                if depth_limit_reached:
+                    log_func('Warning: tree elements were printed up {} depth level only. '
```

Comment:
`tree elements were printed up {} depth level only` -> `the whole hierarchy does not fit into depth={}. `

`You can set larger depth value or use depth=None to print full tree` -> `Increase depth parameter value or set to None (unlimited)`

Reply:
changed